saleor/graphql/shop/types.py

```diff
@@ -66,6 +68,11 @@ class Meta:
 class Shop(graphene.ObjectType):
     available_payment_gateways = graphene.List(
         graphene.NonNull(PaymentGateway),
+        currency=graphene.Argument(
```

Comment:
Not sure of this.
- How storefront will fetch what currencies are available for the shop?
- IMO `PaymentGateway` type should also have a field `currencies` or `supported_currencies`. - It would be easier for storefronts with multiple currencies support.

Reply:
1) There is `shop { currencies }` query. Not sure it will be used in multichannel as well but I'm sure we'll return some channel-specific info so there should be some way for storefront to know the currencies it operates in.
2) Agree that it could be useful, we can discuss with the frontend team if they needed it or look on designs.

Reply:
Remember that someone can use one storefront for multiple currencies. So single storefront- multiple channels. I don't think that extending PaymentGateway would cost us a lot of work, it would be an easier way to fetch all gateways (by storefront) and combine them with currencies.

Reply:
Shouldn't available payment gateways live under Checkout rather than Shop?

Reply:
We have it in both places.

Reply:
We used to have it only in the `Checkout` type but the storefront team asked to add it here as well. @orzechdev Do you remember why we need `availablePaymentGateways` in the `Shop` type?